### PR TITLE
fix libc++ compatibility v2

### PR DESCRIPTION
**Pull request by MrDuartePT**
Fixes https://github.com/hyprwm/hyprlock/issues/73. Tested with Clang/libc++ 17 and 18

### CODE REVIEW

File: src/config/ConfigManager.cpp [hunk 1]
```diff
@@ -3,6 +3,9 @@
 #include <filesystem>
 #include <glob.h>
 #include <cstring>
+#if defined(_LIBCPP_VERSION)
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:6]: why is this guarded?

Comment (MrDuartePT) [reply src/config/ConfigManager.cpp:6]: I notice that it only required for libc++
Well my knowledge of c/c++ is a bit bad, that can be wrong. At least work on my side

Comment (MrDuartePT) [reply src/config/ConfigManager.cpp:6]: Well if yoi think is wrong I can take out

Comment (MrDuartePT) [reply src/config/ConfigManager.cpp:6]: Think about that I should use:
`#if defined(__clang__)` since on gcc it dosent produce a error without mutex

**Comment by MrDuartePT** [comment]
Probably tomorrow I can't change anything until the end of the day. 
So I just commit the change I said above

File: src/config/ConfigManager.cpp (hunk 1 above)
Comment (vaxerski) [reply src/config/ConfigManager.cpp]: just include it without guards

Comment (MrDuartePT) [reply src/config/ConfigManager.cpp]: Ok

Comment (MrDuartePT) [reply src/config/ConfigManager.cpp]: Done

**Review by vaxerski — APPROVED** [review]
cool, thanks!

**Pull request merged**